examples/scripts/iframe/index.mjs

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs';
+import Babel from '@babel/standalone';
+import Handlebars from 'handlebars';
+import formatters from '../../src/app/helpers/formatters.mjs';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const MAIN_DIR = `${__dirname}/../../`;
+
+const EXAMPLE_CONSTS = [
+    "vshader",
+    "fshader",
+    "fshaderFeedback",
+    "fshaderCloud",
+    "vshaderFeedback",
+    "vshaderCloud"
+];
+
+function retrieveConstString(data, name) {
+    const start = data.indexOf(`const ${name} = `);
+    if (start < 1) return;
+    const end = data.indexOf("`;", start);
+    return data.substring(start + name.length + 10, end);
+}
+
+function loadHtmlTemplate(data) {
+    const html = fs.readFileSync(`${MAIN_DIR}/scripts/iframe/index.mustache`, "utf8");
+    const template = Handlebars.compile(html);
+    return template(data);
+}
+
+function buildExample(category, filename) {
+    const exampleString = fs.readFileSync(
+        `${MAIN_DIR}/src/examples/${category}/${filename}`,
+        "utf8"
+    );
+    const exampleConstValues = [];
+    EXAMPLE_CONSTS.forEach((key) => {
+        const value = retrieveConstString(exampleString, key);
+        if (!value) return;
+        exampleConstValues.push({ k: key, v: value });
+    });
```

Comment:
A forEach/if/push is basically a map/filter (which can still infer correct types + less code):

![image](https://user-images.githubusercontent.com/5236548/149922603-8b014021-9187-46f4-aa13-116fa577d6b8.png)

```suggestion
    const exampleConstValues = EXAMPLE_CONSTS
        .map(k => ({k, v: retrieveConstString(exampleString, k)}))
        .filter(_=>_.v);
```

Reply:
Yeah i'm a fan of maps and filters, I should've went with this. Thanks 👍